### PR TITLE
Survival Teams: Death Arena - Initial

### PR DESCRIPTION
**Pull request by arcadeboss**
Survival Teams: Death Arena
- 9 minute "survival" period of armor/weapon resource-collection and crafting
- 3 minute scored Arena battle between the two teams
- Teams spawn in different corners and eventually are teleported to Arena
- A small 350x350 world is created from 3 prebuilt regions and limited generated chunks
- 350x350 block world border, no generated chunks outside border
- Border not escapable by observers, since it has only 2 fixed sizes
- No blitz, players can join at any time with a basic survival kit
- Players keep all weapons, armor and most other items when they die
- As a time-saving buff, the only way to die during "survival" is the void
- All mobs are on outside of Arena
- (level.dat not used for this map)

### CODE REVIEW

File: other/survival/survival_teams_death_arena/map.xml [hunk 1]
```diff
@@ -0,0 +1,467 @@
...
+    <point id="blue-arena-spawn">15.5,55,-15.5</point>
+    <union id="arena-spawn-protect">
+        <rectangle min="-12.5,13.5" max="-17.5,18.5"/> <!-- Red -->
+        <rectangle min="13.5,-12.5" max="17.5,-17.5"/> <!-- Blue -->
+    </union>
+    <apply kit="void-death">
+        <region>
+            <region id="arena-void"/>
+        </region>
+    </apply>
```
Comment (mitchts) [inline other/survival/survival_teams_death_arena/map.xml:135]:
```suggestion
    <apply kit="void-death" region="arena-void"/>
```

File: other/survival/survival_teams_death_arena/map.xml [hunk 2]
```diff
@@ -0,0 +1,467 @@
...
+    <apply block-place="never" region="arena-ceiling" message="You may not build higher!"/>
+    <apply block-place="never" region="arena-no-clutch" message="You may not build this low!"/>
+    <apply enter="all(alive,any(arena,cage))" region="arena-keep-out" message="You may not enter the Arena during Survival phase!"/>
+    <apply block-place="all(any(arena,cage),alive)" block-break="all(any(arena,cage),alive)" region="arena-keep-out" message="You may not edit the Arena during Survival phase!"/>
+</regions>
+<portals>
+    <portal destination="red-survival-spawn" forward="all(survival,alive)" filter="red-only" yaw="@90" pitch="@0"/>
+    <portal destination="blue-survival-spawn" forward="all(survival,alive)" filter="blue-only" yaw="@-90" pitch="@0"/>
+    <portal destination="red-arena-spawn" forward="all(any(arena,cage),alive)" filter="red-only" yaw="@-135" pitch="@0"/>
+    <portal destination="blue-arena-spawn" forward="all(any(arena,cage),alive)" filter="blue-only" yaw="@45" pitch="@0"/>
```
[the remark this answers is not in the document]
Comment (arcadeboss) [reply other/survival/survival_teams_death_arena/map.xml:160]: Ah, that's what that hiccup is.  Thanks, I will look into this.

File: other/survival/survival_teams_death_arena/map.xml [hunk 3]
```diff
@@ -0,0 +1,467 @@
...
+    <cuboid id="cage-region" min="-3,-2,-3" max="3,3,3"/>
+    <cuboid id="space" min="-2,-1,-2" max="2,2,2"/>
+    <translate id="cage-red" offset="-15,56,16"><mirror normal="1,0,0" origin="0,0,0"><region id="cage-region"/></mirror></translate>
+    <translate id="space-red" offset="-15,56,16"><mirror normal="1,0,0" origin="0,0,0"><region id="space"/></mirror></translate>
+    <translate id="cage-blue" offset="16,56,-15"><mirror normal="1,0,1" origin="0,0,0"><region id="cage-region"/></mirror></translate>
+    <translate id="space-blue" offset="16,56,-15"><mirror normal="1,0,1" origin="0,0,0"><region id="space"/></mirror></translate>
+    <apply block-place="never" region="arena-spawn-protect" message="You may not obstruct spawn!"/>
+    <apply enter="never" region="restricted-space" message="You may not leave the map!"/>
+    <apply enter="never" region="red-area" filter="blue-only" message="You may not enter Red's domain!"/>
+    <apply enter="never" region="blue-area" filter="red-only" message="You may not enter Blue's domain!"/>
```
[the remark this answers is not in the document]
Comment (arcadeboss) [reply other/survival/survival_teams_death_arena/map.xml]: Nice, updated.

File: other/survival/survival_teams_death_arena/map.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (arcadeboss) [reply other/survival/survival_teams_death_arena/map.xml]: Yes, the portal for survival phase wasn't needed, removed and it works smoother now.

File: other/survival/survival_teams_death_arena/map.xml (hunk 1 above)
Comment (arcadeboss) [reply other/survival/survival_teams_death_arena/map.xml]: Done

**Pull request merged**